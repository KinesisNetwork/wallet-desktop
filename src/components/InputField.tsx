import * as React from 'react'

export interface Props {
  id: string
  value: string
  label?: string
  icon?: string
  helpText?: string
  placeholder?: string
  type?: 'password' | 'text' | 'number'
  isDisabled?: boolean
  hasButton?: boolean
  isInvalidInput?: boolean
  onChangeHandler: (newValue: string) => any
}

export const InputField: React.SFC<Props> = ({
  value,
  label,
  onChangeHandler,
  type = 'text',
  id,
  helpText,
  placeholder,
  icon,
  hasButton,
  isDisabled,
  isInvalidInput
}) => (
    <div className={`field ${hasButton ? 'control is-expanded' : ''}`}>
      {label && <label className='label is-small'>{label}</label>}
      <div className={`control ${icon ? 'has-icons-left' : ''}`}>
        <input
          className={`input ${isInvalidInput ? 'is-danger' : ''}`}
          id={`input-${id}`}
          onChange={(ev) => onChangeHandler(ev.target.value)}
          type={type}
          placeholder={placeholder}
          value={value}
          disabled={isDisabled}
        />
        {icon && <span className='icon is-left has-text-grey-lighter'><i className={`fas ${icon}`} /></span>}
      </div>
      <p className='help is-link'>
        {helpText}
      </p>
    </div>
  )

InputField.displayName = 'InputField'
