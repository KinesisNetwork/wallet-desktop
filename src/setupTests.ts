import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
