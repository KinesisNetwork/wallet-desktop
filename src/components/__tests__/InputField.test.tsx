import { shallow } from 'enzyme';
import * as React from 'react'

import { InputField } from '@components/InputField'
import '../../setupTests'

describe('InputField', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <InputField
        id='connection-name'
        value=''
        onChangeHandler={() => null}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should display a label if defined in props', () => {
    const wrapper = shallow(
      <InputField
        id='connection-name'
        value=''
        label='Connection Name'
        onChangeHandler={() => null}
      />
    )

    expect(wrapper.children()).toHaveLength(3)
    expect(wrapper.find('label')).toHaveLength(1)
    expect(wrapper.find('label').text()).toEqual('Connection Name')
  })

  it('displays input element with the optional props', () => {
    const wrapper = shallow(
      <InputField
        id='connection-name'
        value=''
        onChangeHandler={() => null}
        type='text'
        placeholder='Enter something'
      />
    )

    const input = wrapper.find('input')

    expect(input.prop('type')).toEqual('text')
    expect(input.prop('placeholder')).toEqual('Enter something')
  })

  it('renders an icon if defined in props', () => {
    const wrapper = shallow(
      <InputField
        id='connection-name'
        value=''
        onChangeHandler={() => null}
        icon='fa-plug'
      />
    )

    const icon = wrapper.find('i')

    expect(icon).toHaveLength(1)
    expect(icon.hasClass('fa-plug')).toBe(true)
  })

  it('renders a paragparh with help text if it is defined in props', () => {
    const wrapper = shallow(
      <InputField
        id='connection-name'
        value=''
        onChangeHandler={() => null}
        helpText='Help'
      />
    )

    const helpText = wrapper.find('.help')
    expect(helpText).toHaveLength(1)
    expect(helpText.text()).toEqual('Help')
  })

  it('calls the onChangeHandler method on values entered', () => {
    const onChangeHandlerMock = jest.fn()
    const wrapper = shallow(
      <InputField
        id='connection-name'
        value=''
        onChangeHandler={onChangeHandlerMock}
        helpText='Help'
      />
    )

    const input = wrapper.find('input')
    input.simulate('change', { target: { value: 'e' } })
    expect(onChangeHandlerMock).toHaveBeenCalledWith('e')
  })
})
