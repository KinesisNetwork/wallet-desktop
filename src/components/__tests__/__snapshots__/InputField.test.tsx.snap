// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should render correctly 1`] = `
<div
  className="field "
>
  <div
    className="control "
  >
    <input
      className="input "
      id="input-connection-name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <p
    className="help is-link"
  />
</div>
`;
